source/api/plugins/before-spec-api.md

```diff
@@ -0,0 +1,65 @@
+---
+title: before:spec
+---
+
+The `before:spec` event fires before a spec file is run.
+
+The event timing is different depending on if Cypress is run via `cypress run` or `cypress open`.
+
+- During `cypress run`, the event is fired as soon as possible when a spec is about to run.
+- During `cypress open`, since there is not a discreet before or after to a spec running, the event is fired once before the browser is launched for a spec.
```

Comment:
Would like some clarification on how this works when running 'Run all intergration specs / run all component tests'